Example/src/LightboxExample.js

```diff
@@ -0,0 +1,259 @@
+import React, { useState, useRef } from 'react';
+import Animated, {
+  useSharedValue,
+  useAnimatedStyle,
+  runOnUI,
+  useAnimatedGestureHandler,
+  interpolate,
+  Extrapolate,
+  withTiming,
+  Easing,
+} from 'react-native-reanimated';
+import { Dimensions, StyleSheet, View, Image } from 'react-native';
+import {
+  ScrollView,
+  PanGestureHandler,
+  TouchableWithoutFeedback,
+} from 'react-native-gesture-handler';
+import { Header } from 'react-navigation-stack';
+
+const AnimatedImage = Animated.createAnimatedComponent(Image);
+
+const dimensions = Dimensions.get('window');
+const GUTTER_WIDTH = 3;
+const NUMBER_OF_IMAGES = 4;
+const IMAGE_SIZE =
+  (dimensions.width - GUTTER_WIDTH * (NUMBER_OF_IMAGES - 1)) / NUMBER_OF_IMAGES;
+
+const styles = StyleSheet.create({
+  container: {
+    paddingTop: 0,
+  },
+
+  scrollContainer: {
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+  },
+
+  backdrop: {
+    ...StyleSheet.absoluteFillObject,
+    backgroundColor: 'black',
+  },
+});
+
+function ImageList({ images, onItemPress }) {
+  return (
+    <ScrollView contentContainerStyle={styles.scrollContainer}>
+      {images.map((item, i) => (
+        <ListItem onPress={onItemPress} key={i} index={i} item={item} />
+      ))}
+    </ScrollView>
+  );
+}
+
+function ListItem({ item, index, onPress }) {
+  const ref = useRef();
+  const opacity = useSharedValue(1);
+
+  const containerStyle = {
+    marginRight: (index + 1) % 4 === 0 ? 0 : GUTTER_WIDTH,
+    marginBottom: GUTTER_WIDTH,
+  };
+
+  const styles = useAnimatedStyle(() => {
+    return {
+      width: IMAGE_SIZE,
+      height: IMAGE_SIZE,
+      opacity: opacity.value,
+    };
+  });
+
+  return (
+    <TouchableWithoutFeedback
+      style={containerStyle}
+      onPress={() => onPress(ref, item, opacity)}>
+      <AnimatedImage ref={ref} source={{ uri: item.uri }} style={styles} />
+    </TouchableWithoutFeedback>
+  );
+}
+
+function useSharedVector(x, y) {
```

Comment:
both `useSharedVector` and `timingConfig` are used only in `ImageTransition` so they could be declared inside that component

Reply:
I got rid of useSharedVector.

But for timingConfig, I prefer static things to stay static. We don't really need to recreate it for each render :)